### PR TITLE
Make two-factor auth optional

### PR DESCRIPTION
**Pull request by omarcopires**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed 
Using two-factor authentication is now optional.

Cherry-pick: https://github.com/Zbizu/forgottenserver/commit/07e66f7be06e5ee8c37aed1e387d705eb022d0a2

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by Zbizu** [comment]
Context behind this change: most version 12 tools use login/password sessionKey without including two more fields for token and timestamp. This causes server to refuse the connection. This change allows disabling check for these two fields from config.lua, making the server more compatible with these tools.

**Review by DSpeichert — CHANGES_REQUESTED** [review]
You're changing default behavior - introducing new config option with a default value opposite to the current behavior.

Also, cherry-picking refers to (typically) backporting changes from one newer code to previous, maintained branches. This is cherry-picking from... where exactly?

**Comment by omarcopires** [comment]
> You're changing default behavior - introducing new config option with a default value opposite to the current behavior.
> 
> Also, cherry-picking refers to (typically) backporting changes from one newer code to previous, maintained branches. This is cherry-picking from... where exactly?

hello, the code was removed from https://github.com/Zbizu/forgottenserver/commit/07e66f7be06e5ee8c37aed1e387d705eb022d0a2

File: src/configmanager.cpp [hunk 1]
```diff
@@ -248,6 +248,7 @@ bool ConfigManager::load()
 	boolean[ONLY_INVITED_CAN_MOVE_HOUSE_ITEMS] = getGlobalBoolean(L, "onlyInvitedCanMoveHouseItems", true);
 	boolean[REMOVE_ON_DESPAWN] = getGlobalBoolean(L, "removeOnDespawn", true);
 	boolean[PLAYER_CONSOLE_LOGS] = getGlobalBoolean(L, "showPlayerLogInConsole", true);
+	boolean[TWO_FACTOR_AUTH] = getGlobalBoolean(L, "enableTwoFactorAuth", false);
```
Comment (Zbizu) [inline src/configmanager.cpp:251]: set here to `true` and the default behaviour will be kept

**Review by Zbizu — CHANGES_REQUESTED** [review]

**Comment by Zbizu** [comment]
> Also, cherry-picking refers to (typically) backporting changes from one newer code to previous, maintained branches. This is cherry-picking from... where exactly?

It comes from a fork that is a few features ahead of the main repo so as long as it doesn't conflict it's just as you described.

**Review by EPuncker — APPROVED** [review]

**Comment by EPuncker** [comment]
@DSpeichert this can be accepted/merged but your request change status is blocking it

**Review by nekiro — APPROVED** [review]

**Review by Zbizu — APPROVED** [review]

**Review by EPuncker — APPROVED** [review]

**Pull request merged**